wakuv2/waku.go

```diff
@@ -1191,16 +1213,52 @@ func (w *Waku) checkForConnectionChanges() {
 		w.onPeerStats(latestConnStatus)
 	}
 
+	w.ConnectionChanged(connection.State{
+		Type:    w.state.Type, //setting state type as previous one since there won't be a change here
+		Offline: !latestConnStatus.IsOnline,
+	})
+}
+
+func (w *Waku) reportPeerMetrics() {
 	if w.statusTelemetryClient != nil {
 		connFailures := FormatPeerConnFailures(w.node)
 		w.statusTelemetryClient.PushPeerCount(w.ctx, w.PeerCount())
 		w.statusTelemetryClient.PushPeerConnFailures(w.ctx, connFailures)
-	}
 
-	w.ConnectionChanged(connection.State{
-		Type:    w.state.Type, //setting state type as previous one since there won't be a change here
-		Offline: !latestConnStatus.IsOnline,
-	})
+		peerCountByOrigin := make(map[wps.Origin]uint)
+		peerCountByShard := make(map[uint16]uint)
+		for _, peerID := range w.node.Host().Network().Peers() {
+			wakuPeerStore := w.node.Host().Peerstore().(wps.WakuPeerstore)
+
+			origin, err := wakuPeerStore.Origin(peerID)
+			if err != nil {
+				continue
+			}
```

Comment:
Should a failutre to retrieve the origin of a peer stop the retrieval of the peer pubsub topics in line 1239?

Reply:
Perhaps not, this seems like an edge case we might want to catch